code/game/antagonist/_antagonist_setup.dm

```diff
@@ -67,16 +67,22 @@ var/global/list/bantype_to_antag_age = list()
 		return antag.current_antagonists
 	return list()
 
-/proc/player_is_antag(var/datum/mind/player, var/only_offstation_roles = 0)
+/proc/player_is_antag(var/datum/mind/player, var/only_offstation_roles = FALSE, var/return_antag_datum = FALSE)
 	for(var/antag_type in all_antag_types)
 		var/datum/antagonist/antag = all_antag_types[antag_type]
 		if(only_offstation_roles && !(antag.flags & ANTAG_OVERRIDE_JOB))
 			continue
 		if(player in antag.current_antagonists)
-			return 1
+			if(return_antag_datum)
+				return antag
+			return TRUE
 		if(player in antag.pending_antagonists)
-			return 1
-	return 0
+			if(return_antag_datum)
+				return antag
+			return TRUE
+	if(return_antag_datum)
+		return null
+	return FALSE
```

Comment:
This is really awkward. You should set things up such that this returns TRUE/FALSE in terms of a datum/null.